src/read/read_options.rs

```diff
@@ -0,0 +1,119 @@
+use crate::StrictnessLevel;
+
+use super::general::{open_with_options, ReadResult};
+
+/// Used to set which format to read the file in.
+#[derive(Debug, Clone, Copy, Default)]
+pub enum Format {
+    /// Load PDB files
+    Pdb,
+    /// Load mmCIF files
+    Mmcif,
+    /// Automatically detect the format
+    #[default]
+    Auto,
+}
+
+impl From<&str> for Format {
+    fn from(s: &str) -> Self {
+        match s {
+            "pdb" => Self::Pdb,
+            "mmcif" => Self::Mmcif,
+            _ => panic!("Unknown format: {}", s),
+        }
+    }
+}
+
+/// Options and flags which can be used to configure how a structure file is
+/// opened.
+///
+/// This builder exposes the ability to configure how a [`PDB`] is loaded.
+///
+/// Generally speaking, when using `ReadOptions`, you'll first call
+/// [`ReadOptions::new`], then chain calls to methods to set each option, then
+/// call [`ReadOptions::read`].
+///
+/// # Examples
+///
+/// Opening a file to read:
+///
+/// ```no_run
+/// use pdbtbx::*;
+///
+/// let pdb = ReadOptions::new()
+///     .set_format(Format::Auto)
+///     .set_level(StrictnessLevel::Loose)
+///     .set_discard_hydrogens(true)
+///     .read("1CRN.pdb");
+//
+/// ```
+#[derive(Debug, Default)]
+pub struct ReadOptions {
+    /// The format to read the file in.
+    pub(crate) format: Format,
+
+    /// The strictness level to use when reading the file.
+    pub(crate) level: StrictnessLevel,
+
+    /// Controls whether to capitalise the chains in the structure.
+    pub(crate) capitalise_chains: bool,
+
+    /// Decompress
+    #[cfg(feature = "compression")]
+    pub(crate) decompress: bool,
+
+    /// Discard hydrogens
+    pub(crate) discard_hydrogens: bool,
+
+    /// Only read the first model
+    pub(crate) only_first_model: bool,
```

Comment:
Do you think a flag to auto renumber any file you open could be useful?

Reply:
Yea thats a good idea. I think with this approach, we should be able to easily add fields to the builder pattern without disrupting the API. If you are happy with the overall API then we should merge this.

We can other parameters later if we want to. What other things would be good to have?

Reply:
Yes that is a very good reason to use this api. I will merge for now then we can add further things later. I do not have any other ones then renumbering in my head.